File: 0041-first-missing-positive/0041-first-missing-positive.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        numSet = set()
        n = max(nums)
        for num in nums:
            numSet.add(num)

        limit = 2**31
        for i in range(1,limit):
            if i not in numSet:
                return i

File: 0041-first-missing-positive/0041-first-missing-positive.java
class Solution {
    public int firstMissingPositive(int[] nums) {
        Set<Integer> s = new HashSet<>();
        for (int num : nums) {
            s.add(num);
        }
        for (int i = 1; i <= Integer.MAX_VALUE; i++) {
            if (!s.contains(i))
                return i;
        }
        return -1;
    }
}